### ISSUE TITLE
Interceptors can cause an invocation of missing partial methods

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#70161]
**Version Used**: 
```
Microsoft Visual Studio Professional 2022
Version 17.8.0 Preview 2.0
VisualStudio.17.Preview/17.8.0-pre.2.0+34112.27
Microsoft .NET Framework
Version 4.8.09037
```

**Steps to Reproduce**:

Compile and run the following code
```cs
using System.Runtime.CompilerServices;
using System;

var c = new C();
c.TestMethod();

partial class C
{
  [InterceptsLocation("C:\\Work\\TestNet7App\\ConsoleApp1\\Program.cs", 12, 31)]
  partial void M();

  public void TestMethod() => Interceptable();
  public void Interceptable() => Console.WriteLine("original");
}

namespace System.Runtime.CompilerServices
{
  [AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
  public class InterceptsLocationAttribute : System.Attribute
  {
    public InterceptsLocationAttribute(string path, int line, int character) { }
  }
}
```

**Expected Behavior**:
Either an error is reported for an interceptor without an implementation part, or the call is not intercepted at runtime.

**Actual Behavior**:
Runtime exception is thrown due to an invocation of a non-existing method: 
```
System.MissingMethodException: 'Method not found: 'Void C.M()'.'
```

**Comment by RikkiGibson** [comment]
I think the correct resolution is to report an error here, as replacing with a method whose calls are omitted would result in too much of a change in semantics, e.g. not evaluating the method's arguments.
